subprojects/core/src/main/java/org/gradle/invocation/DefaultGradle.java

```diff
@@ -209,11 +210,11 @@ public void setRootProject(ProjectInternal rootProject) {
 
     @Override
     public void rootProject(Action<? super Project> action) {
-        if (rootProjectActions != null) {
-            rootProjectActions.add(action);
-        } else {
+        if (projectsLoaded) {
```

Comment:
⭕️ Instead of adding another field, consider extracting the condition into a method with that name. This way we get the clarity without memory cost.

Reply:
I did consider that. However, I prefer an extra property out of consideration:

- Obviously, it's more clear to use a property named `projectsLoaded`.
- `DefaultGradle` would be instantiated only once (or several times?), I think I can accept the extra bytes.

Reply:
Yes, the overhead in this case would be minimal. However in general Gradle's configuration time performance is highly constrained by memory usage. This is why I wanted to point out the alternative, because it'll be useful in other scenarios.